hicommonwealth/commonwealth
# Conditionally use cacheRefresh for memberships

## Link to Issue
Closes: #5804

## Description of Changes
- Uses cached balances for refresh triggered by group requirements update + gated actions
- Uses fresh balances for single-user refresh (on page refresh)

## Test Plan
CA testing this on demo is not feasible, but locally is doable.

The goal of the test is to confirm that gated actions use cached data while page refreshes use fresh data:
- Set env `MEMBERSHIP_REFRESH_TTL_SECONDS` = 15
- Set env `TBC_BALANCE_TTL_SECONDS` = 1
- As a non-admin, go to a gated thread in which you are not the author
- Truncate the Memberships table (`DELETE FROM “Memberships”`)
- Perform these steps quickly in less than 10 seconds
    - Upvote the thread to create a reaction
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that 1 exists, take note of the `last_checked` time
    - Click the upvote button multiple times to create the reaction multiple times
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that 1 exists, and the `last_checked` column is the SAME as before
- Perform these steps quickly
    - Refresh the page
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that at least 1 exists, take note of the `last_checked` time
    - Confirm that every time the page refreshes, the `last_checked` time is DIFFERENT from before

## Deployment Plan
N/A

## Other Considerations
N/A
